kotlin-analysis-api/src/test/kotlin/com/google/devtools/ksp/impl/test/AbstractKSPAATest.kt

```diff
@@ -156,15 +158,17 @@ abstract class AbstractKSPAATest : AbstractKSPTest(FrontendKinds.FIR) {
         val analysisSession = buildStandaloneAnalysisAPISession(withPsiDeclarationFromBinaryModuleProvider = true) {
             registerOnce<FileAttributeService> { DummyFileAttributeService }
             registerOnce(::ClsKotlinBinaryClassCache)
+            CoreApplicationEnvironment.registerExtensionPoint(
+                project.extensionArea,
+                KtResolveExtensionProvider.EP_NAME.name,
+                KtResolveExtensionProvider::class.java
+            )
             buildKtModuleProviderByCompilerConfiguration(compilerConfiguration)
-            project.extensionArea.apply {
-                registerExtensionPoint(
-                    KtResolveExtensionProvider.EP_NAME.name,
-                    KtResolveExtensionProvider::class.java.name,
-                    ExtensionPoint.Kind.INTERFACE,
-                    false
-                )
-            }
+        }.apply {
+            (project as MockProject).registerService(
+                JvmFirDeserializedSymbolProviderFactory::class.java,
+                JvmFirDeserializedSymbolProviderFactory::class.java
```

Comment:
Any insight on this magic?

Reply:
`FirStandaloneServiceRegistrar` registers `JvmStubBasedDeserializedSymbolProviderFactory`, which is IDE only. Standalone mode should go with `JvmFirDeserializedSymbolProviderFactory` per [this comment](https://github.com/JetBrains/kotlin/blob/583aad23506ef235a6136cab5c540a4d620dd60f/analysis/low-level-api-fir/src/org/jetbrains/kotlin/analysis/low/level/api/fir/stubBased/deserialization/JvmStubBasedFirDeserializedSymbolProvider.kt#L39).